.github/workflows/branch-status-checks.yml

```diff
@@ -0,0 +1,56 @@
+name: Branch Version Status Checks
+
+on:
+  push:
+    branches: [ "main", "7.*", "8.*" ]
+  pull_request:
+    branches: [ "*" ]
+  workflow_call:
+    # Map the workflow outputs to job outputs
+    outputs:
+      branches:
+        description: "List of target branches"
+        value: ${{ jobs.list-target-branches.outputs.matrix }}
+
+jobs:
+  list-target-branches:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.get-branch-list.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v2
+      - id: get-branch-list
+        run: echo "::set-output name=matrix::$(yq '.target_branch_list' ./detection_rules/etc/target-branches.yml)"
```

Comment:
we could also use python here and expose a util command which outputs the same contents from https://github.com/elastic/detection-rules/blob/fd9c9f8abf951b7e759decf350a16dee88445b45/detection_rules/schemas/__init__.py#L258-L268.

something like `python -m detection_rules dev utils get_branches`. This would allow us to maintain the centralization while also avoiding the messy yaml-foo